tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/TypeContext.scala

```diff
@@ -30,16 +30,16 @@ class TypeContext(val largestVarIndex: Int, val types: Map[String, (TlaType1, Se
 object TypeContext {
 
   /**
-   * The value of the variable index that means that all variables can be quantified.
+   * A type scheme that carries a type and a set of variables that are universally quantified.
    */
-  val NIL_VAR_INDEX: Int = -1
+  type TypeScheme = (TlaType1, Set[Int])
```

Comment:
Maybe move this type alias to the top of the file (or at least above where it is first used?

Reply:
For some reason, this does not work in Scala. I guess, type definitions work only in packages. Anyhow, it is a good idea to move it closer to `TlaType1`.

Reply:
I have refactored it to `TlaType1Schema` next to `TlaType1`.